src/gwt/src/org/rstudio/studio/client/workbench/prefs/model/Prefs.java

```diff
@@ -359,10 +359,12 @@ protected void doSetValue(JsObject root, String name, String value)
    public class EnumValue extends JsonValue<String>
    {
       private EnumValue(String name, String title, String description, 
-                        String[] values, String defaultValue)
+                        String[] values, String defaultValue, String[] readableValues)
       {
          super(name, title, description, defaultValue);
          allowedValues_ = values;
+         readableValues_ = readableValues;  // Human readable text describing the readable values
+         assert(allowedValues_.length == readableValues_.length);  // DEBUG: Is this a good idea to validate like this?  Probably need to do an explicit throw instead?
```

Comment:
This is a private constructor so I'd suggest moving this check upstream, and adding a debug.warn statement that logs the enum in question to the console.